modules/javafx.graphics/src/main/native-glass/win/ViewContainer.cpp

```diff
@@ -1260,15 +1260,11 @@ static char * touchEventName(unsigned int dwFlags) {
 
 void NotifyTouchInput(
         HWND hWnd, jobject view, jclass gestureSupportCls,
-        const TOUCHINPUT* ti, unsigned count)
+        const TOUCHINPUT* ti, unsigned count, bool isDirect)
 {
 
     JNIEnv *env = GetEnv();
 
-    // Sets to 'true' if source device is a touch screen
-    // and to 'false' if source device is a touch pad/pen.
-    const bool isDirect = IsTouchEvent();
-
```

Comment:
touchevents in case WM_TOUCH is always false debugging in devices lenovo yoga book and thinkpad x1 fold. Dont know yet if this behaviour change between devices. Common case is "press and hold" not trigger touch events.